File: src/main/java/net/internetisalie/lunar/luacats/lang/lexer/LuaCatsTokenTypes.java
package net.internetisalie.lunar.luacats.lang.lexer;

import com.intellij.psi.TokenType;
import com.intellij.psi.tree.IElementType;
import com.intellij.psi.tree.TokenSet;

public interface LuaCatsTokenTypes {
    IElementType LCATS_DASHES = new LuaCatsElementType("LCATS_DASHES");
    IElementType LCATS_WHITESPACE = new LuaCatsElementType("LCATS_WHITESPACE");
    IElementType LCATS_TEXT = new LuaCatsElementType("LCATS_TEXT");

    IElementType LCATS_NAME = new LuaCatsElementType("LCATS_NAME");
    IElementType LCATS_STRING = new LuaCatsElementType("LCATS_STRING");
    IElementType LCATS_SYMBOL = new LuaCatsElementType("LCATS_SYMBOL");
    IElementType LCATS_TAG = new LuaCatsElementType("LCATS_TAG");
    IElementType LCATS_KEYWORD = new LuaCatsElementType("LCATS_KEYWORD");
    IElementType LCATS_CODE = new LuaCatsElementType("LCATS_CODE");
    IElementType LCATS_NUMBER = new LuaCatsElementType("LCATS_NUMBER");

    IElementType LCATS_BAD_CHARACTER = TokenType.BAD_CHARACTER;

    TokenSet LUACATS_TOKENS = TokenSet.create(
            LCATS_DASHES,
            LCATS_WHITESPACE,
            LCATS_TEXT,
            LCATS_NAME,
            LCATS_STRING,
            LCATS_SYMBOL,
            LCATS_TAG,
            LCATS_KEYWORD,
            LCATS_CODE,
            LCATS_NUMBER
    );
}

File: src/main/kotlin/net/internetisalie/lunar/luacats/lang/lexer/LuaCatsTokenTypes.kt
package net.internetisalie.lunar.luacats.lang.lexer

import com.intellij.psi.TokenType
import com.intellij.psi.tree.IElementType

object LuaCatsTokenTypes {
    @JvmField val LCATS_DASHES: IElementType = LuaCatsElementType("LCATS_DASHES")
    @JvmField val LCATS_WHITESPACE: IElementType = LuaCatsElementType("LCATS_WHITESPACE")
    @JvmField val LCATS_TEXT: IElementType = LuaCatsElementType("LCATS_TEXT")

    @JvmField val LCATS_NAME: IElementType = LuaCatsElementType("LCATS_NAME")
    @JvmField val LCATS_STRING: IElementType = LuaCatsElementType("LCATS_STRING")
    @JvmField val LCATS_SYMBOL: IElementType = LuaCatsElementType("LCATS_SYMBOL")
    @JvmField val LCATS_TAG: IElementType = LuaCatsElementType("LCATS_TAG")
    @JvmField val LCATS_KEYWORD: IElementType = LuaCatsElementType("LCATS_KEYWORD")
    @JvmField val LCATS_CODE: IElementType = LuaCatsElementType("LCATS_CODE")
    @JvmField val LCATS_NUMBER: IElementType = LuaCatsElementType("LCATS_NUMBER")

    @JvmField val LCATS_BAD_CHARACTER: IElementType = TokenType.BAD_CHARACTER
}
